internal/api/versionsapi/cli/latest.go

```diff
@@ -51,8 +51,14 @@ func runLatest(cmd *cobra.Command, _ []string) error {
 		return fmt.Errorf("creating client: %w", err)
 	}
 	defer func() {
-		if err := clientClose(cmd.Context()); err != nil {
-			log.Errorf("Closing versions API client: %v", err)
+		log.Infof("Invalidating cache. This may take some time")
+		err := clientClose(cmd.Context())
+		// Log invalidation errors in any case, but don't overwrite the return error if there is one.
+		if err != nil {
+			log.Errorf("Failed to invalidate cache: %v", err)
+		}
+		if err != nil && retErr == nil {
+			retErr = fmt.Errorf("invalidating cache: %w", err)
```

Comment:
and here too

Might be worth to throw all of this into its own exported function, so we don't have to write the same 6 lines of code multiple times

Reply:
I've removed the extra logging branch as it is redundant with your suggestion. The defer is now too simple to warrant an extra function imo.